trillian.proto

```diff
@@ -22,12 +34,107 @@ enum SignatureAlgorithm {
+  // mismatched proto versions being used. Represents an invalid value.
+  UNKNOWN_TREE_STATE = 0;
+
+  // Active trees are able to respond to both read and write requests.
+  ACTIVE = 1;
+
+  // Frozen trees are only able to respond to read requests, writting to a
+  // frozen tree is forbidden.
+  FROZEN = 2;
+
+  // Tree was been deleted, therefore is invisible and acts similarly to a
+  // non-existing tree for all requests.
+  // A soft deleted tree may be undeleted while the soft-deletion period has not
+  // passed.
+  SOFT_DELETED = 3;
+
+  // A hard deleted tree was been definitely deleted and cannot be recovered.
+  // Acts an a non-existing tree for all read and write requests, but blocks the
+  // tree ID from ever being reused.
+  HARD_DELETED = 4;
+}
+
+// Type of the tree.
+enum TreeType {
+  // Tree type cannot be determined. Included to enable detection of
+  // mismatched proto versions being used. Represents an invalid value.
+  UNKNOWN_TREE_TYPE = 0;
+
+  // Tree represents a verifiable log.
+  LOG = 1;
+
+  // Tree represents a verifiable map.
+  MAP  =2;
+}
+
+// Duplicate policy of a tree.
+enum DuplicatePolicy {
+  // Policy cannot be determined. Included to enable detection of
+  // mismatched proto versions being used. Represents an invalid value.
+  UNKNOWN_DUPLICATE_POLICY = 0;
+
+  // Duplicates are not allowed in the tree and will cause errors on insertion.
+  DUPLICATES_NOT_ALLOWED = 1;
+
+  // Duplicates are allowed in the tree.
+  DUPLICATES_ALLOWED = 2;
+}
+
+// Represents a tree, which may be either a verifiable log or map.
+// Readonly attributes are assigned at tree creation, after which they may not
+// be modified.
+message Tree {
+  // ID of the tree.
+  // Readonly.
+  int64 tree_id = 1;
+
+  // State of the tree.
+  // Trees are active after creation. At any point the tree may transition
+  // between ACTIVE and FROZEN.
+  // Deleted trees are set as SOFT_DELETED for a certain time period, after
+  // which they'll automatically transition to HARD_DELETED.
+  TreeState state = 2;
+
+  // Type of the tree.
+  // Readonly.
+  TreeType tree_type = 3;
+
+  // Hash strategy to be used by the tree.
+  // Readonly.
+  HashStrategy hash_strategy = 4;
+
+  // Hash algorithm to be used by the tree.
+  // Readonly.
+  HashAlgorithm hash_algorithm = 5;
+
+  // Signature algorithm to be used by the tree.
+  // Readonly.
+  SignatureAlgorithm signature_algorithm = 6;
+
+  // Duplicate policy to be used by the tree.
+  // Readonly.
+  DuplicatePolicy duplicate_policy = 7;
+
+  // Display name of the tree.
+  // Optional.
+  string display_name = 8;
+
+  // Description of the tree,
+  // Optional.
+  string description = 9;
+
+  // Timestamp of tree creation.
+  // Readonly.
+  int64 create_time_millis_since_epoch = 10;
+
+  // Timestamp of last tree update.
+  // Readonly (automatically assigned on updates).
+  int64 update_time_millis_since_epoch = 11;
```

Comment:
Would it be useful to have a `repeated Audit` field where the API appends details of any mutations made to this pb?